src/ess/reflectometry/tools.py

```diff
@@ -170,28 +170,43 @@ def _interpolate_on_qgrid(curves, grid):
 
 def scale_reflectivity_curves_to_overlap(
     curves: Sequence[sc.DataArray],
-    return_scaling_factors=False,
-) -> list[sc.DataArray] | list[sc.scalar]:
+    critical_edge_interval: tuple[sc.Variable, sc.Variable] | None = None,
+) -> tuple[list[sc.DataArray], list[sc.scalar]]:
     '''Make the curves overlap by scaling all except the first by a factor.
     The scaling factors are determined by a maximum likelihood estimate
     (assuming the errors are normal distributed).
 
+    If :code:`critical_edge_interval` is provided then all curves are scaled.
+
     All curves must be have the same unit for data and the Q-coordinate.
 
     Parameters
     ---------
     curves:
         the reflectivity curves that should be scaled together
-    return_scaling_factor:
-        If True the return value of the function
-        is a list of the scaling factors that should be applied.
-        If False (default) the function returns the scaled curves.
+    critical_edge_interval: optional,
+        a tuple denoting an interval that is known to belong
+        to the critical edge, i.e. where the reflectivity is
+        known to be 1.
 
     Returns
     ---------
     :
-        A list of scaled reflectivity curves or a list of scaling factors.
+        A list of scaled reflectivity curves and a list of the scaling factors.
     '''
+    if critical_edge_interval is not None:
+        q = next(iter(curves)).coords['Q']
+        N = (
+            ((q > critical_edge_interval[0]) & (q < critical_edge_interval[1]))
+            .sum()
+            .value
+        )
+        edge = sc.DataArray(
+            data=sc.ones(dims=('Q',), shape=(N,), with_variances=True),
+            coords={'Q': sc.linspace('Q', *critical_edge_interval, N + 1)},
+        )
+        curves, factors = scale_reflectivity_curves_to_overlap([edge, *curves])
+        return curves[1:], factors[1:]
```

Comment:
Would it make sense to set the scale factor as a coord in each curve? Returning a tuple of lists seems a bit error prone.

Reply:
Right now I prefer the tuple of lists because it's simple and obvious. For example, setting it as a coord requires the user to know the name of the coord to extract the return value. I don't directly see why it would be more error prone, did you have a particular situation in mind?